File: org.palladiosimulator.blockchainsystems.core/src/main/java/org/palladiosimulator/blockchainsystems/core/system/abstractions/Transaction.java
package org.palladiosimulator.blockchainsystems.core.system.abstractions;

/**
 * The {@code Transaction} interface represents a transaction that can be included in a block.
 *
 * @author Davis Riedel
 */
public interface Transaction {
    /**
     * @return the hash uniquely identifying the transaction
     */
    String getTxId();

    /**
     * @return the maximum amount of gas that can be used by the transaction
     */
    int getGasLimit();

    /**
     * @return the gas price of the transaction in GWEI
     */
    int getGasPrice();

    /**
     * @return the gas actually used by the transaction
     */
    int getGasUsed();
}

File: org.palladiosimulator.blockchainsystems.core/src/main/kotlin/org/palladiosimulator/blockchainsystems/core/system/abstractions/Transaction.kt
package org.palladiosimulator.blockchainsystems.core.system.abstractions

interface Transaction : Propagatable {
  /**
   * Returns the unique identifier of the transaction.
   *
   * @return the transaction ID
   */
  fun getTxId(): String

  /**
   * Returns the size of the transaction in bytes.
   *
   * @return the size of the transaction
   */
  fun getSize(): Int

  /**
   * Returns the time when the transaction was created.
   *
   * @return the creation time of the transaction
   */
  fun getCreationTime(): Long
}